code/modules/food_and_drinks/kitchen_machinery/food_cart.dm

```diff
@@ -121,7 +121,7 @@
 					break
 
 	if(href_list["portion"])
-		portion = CLAMP(input("How much drink do you want to dispense per glass?") as num, 0, 50)
+		portion = CLAMP(input("How much drink do you want to dispense per glass?") as num|null, 0, 50)
```

Comment:
will set to 0 if cancelled instead of doing nothing

Reply:
if it's 0 shouldn't it also early return?